src/solvers/smt2/smt2_conv.cpp

```diff
@@ -1809,6 +1809,63 @@ void smt2_convt::convert_expr(const exprt &expr)
       "smt2_convt::convert_expr: `"+expr.id_string()+
       "' is not yet supported");
   }
+  else if(expr.id() == ID_bswap)
+  {
+    if(expr.operands().size() != 1)
+      INVALIDEXPR("bswap gets one operand");
+
+    if(expr.op0().type() != expr.type())
+      INVALIDEXPR("bswap gets one operand with same type");
+
+    // first 'let' the operand
+    out << "(let ((bswap_op ";
+    convert_expr(expr.op0());
+    out << ")) ";
+
+    if(expr.type().id() == ID_signedbv || expr.type().id() == ID_unsignedbv)
+    {
+      const std::size_t width = to_bitvector_type(expr.type()).get_width();
+
+      // width must be multiple of bytes
+      if(width % 8 != 0)
```

Comment:
Could `config.ansi_c.char_width` please be used so that #917 doesn't end up in an endless catch-up game?

Reply:
I did contemplate this; would the semantics of __builtin_bswapX actually change on a machine with chars that are >8 bits?
The gcc manual says "Byte here always means exactly 8 bits."

Reply:
I'd rather insist that `exprt(ID_bswap)` has a semantics that may differ from that of `__builtin_bswapX`; if there are cases where `__builtin_bswapX` does not swap bytes (of some number of bits) then it's up to the front-end to sort this out. If we want `exprt(ID_bswap)` to always swap multiples of 8 bits then we should 1) rename it and 2) change `boolbvt::convert_bswap` to follow this semantics.

The one case (TI C55x) of a non-8-bit-bytes architecture that I know of doesn't have GCC support so `__builtin_bswapX` wouldn't actually occur.

Reply:
Ok, then I'd advocate to make "number of bits in a byte" a parameter, so we don't rely on a global config.

Reply:
That could be done, but then we've still got `byte_{extract,update}` all the way to the back-end, which also need to know about the width of a byte. I'd just like to come up with a consistent solution for all of those.

Reply:
I'd probably go for the same here.